From 50b8600c63c5487e901e2845a0f64d384a65f75d Mon Sep 17 00:00:00 2001
From: Derek Mauro <dmauro@google.com>
Date: Thu, 18 Sep 2025 11:15:32 -0700
Subject: [PATCH] Add rules_cc dependency, required by Bazel going forward

This also adds the dependencies of rules_cc to WORKSPACE.
bzlmod automatically pulls in dependencies.
skylib is removed as it is pulled in by a deps function.

PiperOrigin-RevId: 808659470
Change-Id: Idc41cad7b05019793d4a1898bdb80bc4797da5cf

--- BUILD.bazel
+++ BUILD.bazel
@@ -30,6 +30,8 @@
 #
 #   Bazel Build for Google C++ Testing Framework(Google Test)
 
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
+
 package(default_visibility = ["//visibility:public"])
 
 licenses(["notice"])

--- MODULE.bazel
+++ MODULE.bazel
@@ -52,4 +52,9 @@ bazel_dep(
 )
 
+bazel_dep(
+    name = "rules_cc",
+    version = "0.2.8"
+)
+
 bazel_dep(
     name = "rules_python",

--- googlemock/test/BUILD.bazel
+++ googlemock/test/BUILD.bazel
@@ -30,6 +30,7 @@
 #
 #   Bazel Build for Google C++ Testing Framework(Google Test)-googlemock
 
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_test")
 load("@rules_python//python:defs.bzl", "py_library", "py_test")
 
 licenses(["notice"])

--- googletest/test/BUILD.bazel
+++ googletest/test/BUILD.bazel
@@ -30,6 +30,7 @@
 #
 # Bazel BUILD for The Google C++ Testing Framework (Google Test)
 
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_test")
 load("@rules_python//python:defs.bzl", "py_library", "py_test")
 
 licenses(["notice"])
